### PR TITLE
Updates to pocketknife

### PR DESCRIPTION
**Pull request by njmyers**
Made a couple of opinionated updates but I think the result is more idiomatic rust code. Let me know if you don't agree with any choices!

- Use result types instead of eprintln and process exit code
- Propagate result types back up to main using `?` operator
- Add `serde` for serialization and deserialization of payloads and responses
- Add `clap` cli parser as it handles env, args and gives us a help menu
- Added some tests
- Added model name as an argument/env variable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,9 +183,7 @@ $ ./yaml2json.sh sample.yaml > sample.json
 1. Install Rust from https://www.rust-lang.org/ if you don't have it already.
 2. Clone this repo locally.
 3. Grab your OpenAI API key and set the `OPENAI_API_KEY` environment variable with it.
-4. Run `cargo build --release` to compile the release binary to `target/release/pk`.
-5. Move the binary to wherever you like and add it to your `PATH`.
-5. Run the `pk` binary using the use cases and examples above for guidance.
+4. Run `cargo install --path .`
```
Comment (njmyers) [inline README.md:186]: This lets cargo manage PATH.

File: src/main.rs [hunk 2]
```diff
@@ -1,117 +1,30 @@
...
+
+    let response = chat::post(&chat::Args {
+        message: &message,
+        model_name: &cli.model_name,
+        api_key: &cli.api_key,
+        timeout: &cli.timeout,
+        debug: &cli.debug,
+    })?;
+
+    if cli.in_place && cli.input.is_some() {
```
Comment (RGBz) [inline src/main.rs:23]: Should error if in place is specified, but no file is specified

File: README.md (hunk 1 above)
Comment (RGBz) [reply README.md:186]: This will install it just in this directory right? Like it won't add it to your PATH?

File: src/chat.rs [hunk 3]
```diff
@@ -0,0 +1,217 @@
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+use crate::error::AnyError;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct Usage {
+    pub prompt_tokens: i64,
```
Comment (RGBz) [inline src/chat.rs:8]: i32 would hopefully suffice or someone is gonna have a hefty bill!

File: src/chat.rs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        match model {
+            "gpt-3.5-turbo" => Ok(Self::Gpt3Turbo),
+            "gpt-4" => Ok(Self::Gpt4),
+            _ => Err("Error: Not a valid model name".into()),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct MessageRequest {
```
Comment (RGBz) [inline src/chat.rs:92]: This type is the same as the response one. Are they broken out to support a future where they might differ?

**Review by RGBz — APPROVED** [review]
Awesome works great! Added a few minor comments, but feel free to merge! Love the extra options and more ergonomic cli experience!

**Comment by RGBz** [comment]
Actually I'm just gonna merge as is

File: README.md (hunk 1 above)
Comment (njmyers) [reply README.md:186]: It will add it to your PATH!

Comment (RGBz) [reply README.md:186]: Yeah I dunno what happened when I tried this a few nights ago. I just tried it again and it worked. Very cool!

**Pull request merged**